### PR TITLE
Implement the virtual input system

### PR DESCRIPTION
**Pull request by MCPfannkuchenYT**
This will be the virtual input system for both the keyboard and the mouse.
German: https://1drv.ms/p/s!Av_ysXerhm5Cphi9RMqnIbvI5Dkv?e=9zfB3C
English: https://1drv.ms/p/s!Av_ysXerhm5CphxfXLbbOwTL-sJo?e=3M5RdS

### CODE REVIEW

**Comment by MCPfannkuchenYT** [comment]
Feel free to merge after testing.

File: src/com/minecrafttas/tasmod/client/virtualinputs/VirtualKeyboard.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		if (hasNext) {
+			VirtualKeyboard.eventKeyCode = Keyboard.getEventKey();
+			VirtualKeyboard.eventKeyState = Keyboard.getEventKeyState();
+			VirtualKeyboard.eventKeyCharacter = Keyboard.getEventCharacter();
+		}
+		
+		/* Input Source 2 (example): The playback file. Add custom input packet sources here into an else block and update hasNext for future input sources. */
+		
+		TASmod.LOGGER.debug("Processed next keyboard event");
+		if (hasNext) VirtualKeyboard.keyStates[VirtualKeyboard.eventKeyCode] = VirtualKeyboard.eventKeyState; // Update the keyboard replica
```
Comment (ScribbleLP) [inline src/com/minecrafttas/tasmod/client/virtualinputs/VirtualKeyboard.java:54]: I see potential problems in this implementation...  
It only updates the necessary keybindings and is an exact mirror of the vanilla keybindings.
The problem comes when you want to load different inputs (e.g. in a savestate) into the keyboard, desyncing the virtual keyboard from the vanilla keyboard.  
This system will then require you to also update the vanilla keyboard when loading a savestate so it doesn't desync.
Not something to fix, but hopefully useful in the future

File: src/com/minecrafttas/tasmod/client/virtualinputs/VirtualMouse.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	/**
+	 * This is the y coordinate of the mouse
+	 */
+	private static int y;
+	
+	/**
+	 * This is the delta x coordinate of the mouse updating every tick
+	 */
+	@SuppressWarnings("unused") // Will be used for playback later...
+	private static int dx;
```
Comment (ScribbleLP) [inline src/com/minecrafttas/tasmod/client/virtualinputs/VirtualMouse.java:47]: 😥

**Review by ScribbleLP — APPROVED** [review]
Looks really similar to the TASmod before the big rewrite... I am not 100% satisfied tho... Well I can't seem to stop you now

**Comment by ScribbleLP** [comment]
Also the problem with making a simpler virtual mouse is that the playback file is going to be massive again...

**Comment by MCPfannkuchenYT** [comment]
We'll see

**Pull request merged**